### ISSUE TITLE
Improved Language Selector

### ISSUE THREAD

**Issue by Volkmire** [futo-org/android-keyboard#681]
Current language selection is not pleasant to use. It requires 2 clicks and moving the finger for more than half of the screen.

Current:
- hold space bar to open a pop up in the middle of the screen
- move finger to the middle of the screen
- click on your language you need to use

Proposed:
- hold space bar to open a scrolling list that extends up from space bar position (meaning your thumb is already on the bottom entry of the list)
- slide the finger on required language
- release finger to select a language (releasing outside of list boundaries does not change the language)

related: futo-org/android-keyboard#85

**Comment by merlinuwe** [comment]
Please consider: When one uses 2 languages he wants to quickly change between them. For only this use case, long press should switch immediately to the other language.

When someone uses 2 languages he may use nonetheless the _same_ keyboard (which he is most comfortable with). But once you selected for example the german keyboard for the german language you cannot use the german language also for the english language.

The German keyboard misses the 7 and the 9 in the first row. Instead it holds the umlauts ü and ö. But ü and ö are already in the keyboard at their correct positions available.

The same is with key a. It shows by long pressing ä, which is also already perfect available. Give it the @ sign.

I miss the sign for Euro € as a second key. I suggest d which currently holds $.

(Sorry for the long critic, which stands in no relation to the great overall quality of your keyboard app.)

**Comment by asdhuman-dev** [comment]
Please add an option to replace emojis button with language switch one too 🙏

**Comment by pvphome** [comment]
> Please add an option to replace emojis button with language switch one too 🙏

Yes, that would be very convenient. The location of the language switch button in the bottom row is used in many other keyboards. Non-standard location of the button in the Futo is simply confusing for a user accustomed to switching languages ​​with a button.
